lib/libspl/timestamp.c

```diff
@@ -51,7 +51,7 @@ print_timestamp(uint_t timestamp_fmt)
 		fmt = nl_langinfo(_DATE_FMT);
 
 	if (timestamp_fmt == UDATE) {
-		(void) printf("%ld\n", t);
+		(void) printf("%" ZFS_TIME_T_FORMAT "d\n", t);
```

Comment:
The way we've handled this elsewhere in the code is by simply casting it as you suggested.  Let's do that here and in `libzfs_sendrecv` as well.

```suggestion
		(void) printf("%lld\n", (longlong_t)t);
```

Reply:
Done